### DISCUSSION TITLE
Unity Catalog

### DISCUSSION THREAD

**Discussion by giovannipapini-agilelab** [apache/superset#23736] [Q&A / Help]
Is is possible to interact with Databricks `Unity Catalog` catalogs? As far as I have seen I currently cannot see catalogs and their databases from a standard databricks connection.

Is there any plan to integrate it?

**Comment by alldoami** [comment]
@giovannipapini-agilelab I don't think they've worked on it yet right? Have you been able to query unity catalog data?

**Comment by giovannipapini-agilelab** [comment]
Hi @alldoami, no I can confirm there is no news about UI integration atm (mainly thinking about a "select catalog" textbox when getting a new dataset for example). But I actually managed to query Unity Catalog tables, only using the pyodbc connector with the Databricks ODBC driver. In that case you can also specify a { "catalog": "main" } parameter in the query engine options to have the "main" catalog when listing databases, but you can also query any other catalog using SQLLab.

**Comment by YiouZuo** [comment]
Hi @giovannipapini-agilelab ,
I ran into the same issue. I was able to query Unity Catalog tables in SQLlab, but neither SQLlab nor Dataset tab would show the correct schema list. 
Turns out that I was using the wrong type of database -- Databricks Interactive Cluster.
So here is what you need to do:
1. Choose `Databricks` when you add a database. (not Databricks Interactive Cluster)
2. Use the following SQLAlchemy URI:
``databricks://token:{access_token}@{server_hostname}:{port}/?&catalog={catalog}&schema={schema}``
3. add the following configuration to “Other” -> “Engine Parameters”, with your HTTP path:
`{"connect_args": {"http_path": "sql/protocolv1/o/****"}}`

**Reply by prakashpatidarcoindcx** [reply]
Sorry, this is not working to explore unity catalog schema in Sql Lab for me.I query but not able to explore schema and shcema showing is hive metastore schema only.

**Comment by kenniy** [comment]
This should work:
* Within Databricks, create a SQL Warehouse if you don't have one. You will need the ID.
* Within your superset setup, ensure you have `apache-superset[databricks]`. See [how](https://superset.apache.org/docs/databases/docker-add-drivers/).
* When connecting, from the **SUPPORTED DATABASES** dropdown, select **Databricks SQL Endpoint**.
* In SQLALCHEMY URI, use `databricks+connector://token:<your_token>@<databricks_hostname>?http_path=/sql/1.0/warehouses/<sql_wh_id>&catalog=<databricks_catalog_name>`
* Go to "ADVANCED" -> "Other" -> "Engine Parameters", and add 
```
{
    "connect_args":
    {
        "http_path": "/sql/1.0/warehouses/<sql_wh_id>",
        "catalog": "catalog_name>"
    }
}
``` 
* Click **CONNECT**

**Reply by jijosg** [reply]
Thanks, this works like a charm!

**Comment by vastnight** [comment]
kenniy's answer above doesn't seem to work for me. if you dont have alot of datasets though you can edit your dataset > virtual sql > in the sql query editor select your table with the right catalog

**Comment by drummerwolli** [comment]
to wrap this up, this is implemented natively in Superset now since some time already: https://github.com/apache/superset/issues/22862
